packages/handler-kit-azure-func/src/function.ts

```diff
@@ -120,12 +120,29 @@ export const httpAzureFunction =
     h: H.Handler<H.HttpRequest, H.HttpResponse<unknown, H.HttpStatusCode>, R>
   ) =>
   (deps: Omit<R, "logger" | "input">) =>
-  (input: unknown, ctx: InvocationContext) =>
+  (request: HttpRequest, ctx: InvocationContext) =>
     pipe(
-      azureFunctionTE(h, {
-        ...deps,
-        inputDecoder: HttpRequestFromAzure,
-      })(input, ctx),
+      TE.tryCatch(
+        () => request.json(),
+        () => new Error("The request body is not a valid JSON.")
+      ),
+      TE.orElse(() => TE.right<Error, unknown>(undefined)),
+      TE.chain((body) =>
+        azureFunctionTE(h, {
+          ...deps,
+          inputDecoder: HttpRequestFromAzure,
+        })(
+          {
+            body,
+            url: request.url,
+            params: request.params,
+            headers: Object.fromEntries(request.headers.entries()),
```

Comment:
I'd add a comment about why we can't just assign the properties

Reply:
is not clear from source types? `HttpRequest`'s `headers` is an `Headers` object, but `handler-kit` requires a `Record<string, string>`. Same for `query` (`URLSearchParams` vs `Record`)